enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,571 @@
+during an upgrade too.
+
+7. The Ingress Operator creates and manages the LB for *.apps by starting a k8s
+Service of type LoadBalancer (existing functionality). Updating this Service to
+accept already created LBs is an upstream change that we are not pursuing at
+this time.
+
+8. When the LoadBalancer Service successfully creates the Ingress LB in the
+underlying cloud, it updates the `Hostname` (in AWS) or `IP` (Azure and GCP)
+fields within the [LoadBalancerIngress struct] (https://github.com/kubernetes/api/blob/cdff1d4efea5d7ddc52c4085f82748c5f3e5cc8e/core/v1/types.go#L4554). These fields represent the hostname
+or the IP address of the Ingress LB.
+
+9. OpenShift needs to provide an in-cluster DNS solution for *.apps resolution
+because several operators depend on this service being available during cluster
+installation. To achieve this, MCO will start a static CoreDNS pod to provide
+DNS resolution for *.apps only.
+
+10. Configuration for the CoreDNS pod is provided by its CoreFile that contains
+the IP address for the Ingress LB. This IP address is obtained by resolving the
+`Hostname` field within the LB service status.
+
+10. runtimecfg monitors the LB service status for changes to the LB IP address
+(by resolving the `Hostname`). If there is a change, it will update the
+Corefile with the new IP address of the Ingress LB.
+
+11. Once the cluster installation is complete, the customer can add a DNS entry
+in their custom DNS solution for *.apps.
+
+12. During cluster deletion, OpenShift is responsible only for deleting the
+Ingress LB and the customer is responsible for deleting the LBs for API and
+API-Int. The customer would also be responsible for removing entries from
+their custom DNS.
+
+#### Variation [optional]
+
+ARO was able to solve this problem in a neat way by making use of the
+[`LoadBalancerIP`] (https://github.com/kubernetes/api/blob/cdff1d4efea5d7ddc52c4085f82748c5f3e5cc8e/core/v1/types.go#L4721) field of the LoadBalancer service manifest.
+
+By using a pre-defined static IP as input to the LoadBalancer Service with the
+same name and annotations as one created by the cluster-ingress-operator, ARO
+is able to create an Ingress LB during the bootstrap phase. The non-cloud DNS
+could be pre-configured to resolve *.apps service to this pre-defined static
+IP.
+
+This would work well except for the following 2 reasons:
+1. AWS does not refer to the `LoadBalancerIP` input value before it creates an
+Ingress LB. So, the above solution only would work for Azure and GCP and not
+for AWS.
+
+2. Based on [K8s API documentation] (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#servicespec-v1-core), the `LoadBalancerIP` field would soon be deprecated.
+
+### API Extensions
+
+1. Install config is augmented to allow the customer to specify if a custom
+DNS solution is being used. In addition, the customer would be able to indicate
+the LB type to be used for API and API-Int traffic.
+
```

Comment:
Where does this fall in the workflow described above? This isn't clear to me.

Reply:
At Step 3.